pages/_app.tsx

```diff
@@ -1,12 +1,18 @@
+import { Analytics } from '@vercel/analytics/react';
 import { SiteProvider } from '../providers/siteProvider';
 import { LocaleProvider } from '../providers/localeProvider';
 import { BlogDataProvider } from '../providers/blogDataProvider';
 import { NodeReleasesProvider } from '../providers/nodeReleasesProvider';
 import { sourceSans } from '../util/nextFonts';
+import * as nextConstants from '../next.constants.mjs';
 import type { AppProps } from 'next/app';
 
 import '../styles/old/index.scss';
 
+// The Vercel Analytics component should only be rendered if the current App is
+// deployed within Vercel Infrastructure. Otherwise we don't need to render this component.
+const renderVercelAnalytics = nextConstants.VERCEL_DEPLOYED && <Analytics />;
```

Comment:
mostly just curious, why here and then on line 28 as the expression?

Reply:
In theory it wouldm only be evaluated during build time, but I guess since WebPack injects those, it doesn't matter, so we can just move to line 28.